libraries/AP_Landing/AP_Landing_Deepstall.cpp

```diff
@@ -0,0 +1,380 @@
+            return false;
+        }
+        type_deepstall_stage = DEEPSTALL_STAGE_ARC;
+        //fallthrough
+    case DEEPSTALL_STAGE_ARC:
+        {
+        Vector2f groundspeed = ahrs.groundspeed_vector();
+        if (!nav_controller->reached_loiter_target() ||
+            (fabsf(wrap_180(type_deepstall_target_heading_deg -
+                            degrees(atan2f(-groundspeed.y, -groundspeed.x) + M_PI))) >= 10.0f)) {
+            nav_controller->update_loiter(type_deepstall_arc, aparm.loiter_radius, 1);
+            return false;
+        }
+        type_deepstall_stage = DEEPSTALL_STAGE_APPROACH;
+        //fallthrough
+        }
+    case DEEPSTALL_STAGE_APPROACH:
+        nav_controller->update_waypoint(type_deepstall_arc_exit, type_deepstall_extended_approach);
+
+        travel_distance = type_deepstall_predict_travel_distance(ahrs.wind_estimate(), (current_loc.alt - ahrs.get_home().alt) / 100.0f);
+
+        memcpy(&entry_point, &type_deepstall_landing_point, sizeof(Location));
+        location_update(entry_point, type_deepstall_target_heading_deg + 180.0, travel_distance);
+
+        if (!location_passed_point(current_loc, type_deepstall_arc_exit, entry_point)) {
+            if (location_passed_point(current_loc, type_deepstall_arc_exit, type_deepstall_extended_approach)) {
+                // this should never happen, but prevent against an indefinite fly away
+                type_deepstall_stage = DEEPSTALL_STAGE_FLY_TO_LANDING;
+            }
+            return false;
+        }
+        type_deepstall_stage = DEEPSTALL_STAGE_LAND;
+        type_deepstall_stall_entry_time = AP_HAL::millis();
+        elevator = SRV_Channels::get_channel_for(SRV_Channel::k_elevator);
+        if (elevator != nullptr) {
+            // take the last used elevator angle as the starting deflection
+            // don't worry about bailing here if the elevator channel can't be found
+            // that will be handled within override_servos
+            type_deepstall_initial_elevator_pwm = elevator->get_output_pwm();
+        }
+        type_deepstall_l1_xtrack_i = 0; // reset the integrators
+        //fallthrough
+    case DEEPSTALL_STAGE_LAND:
+        // while in deepstall the only thing verify needs to keep the extended approach point sufficently far away
+        nav_controller->update_waypoint(current_loc, type_deepstall_extended_approach);
+        return false;
+    default:
+        return true;
+    }
+}
+
+bool AP_Landing::type_deepstall_override_servos(void)
+{
+    if (!(type_deepstall_stage == DEEPSTALL_STAGE_LAND)) {
+        return false;
+    }
+
+    SRV_Channel* elevator = SRV_Channels::get_channel_for(SRV_Channel::k_elevator);
+    SRV_Channel* aileron = SRV_Channels::get_channel_for(SRV_Channel::k_aileron);
+    SRV_Channel* rudder = SRV_Channels::get_channel_for(SRV_Channel::k_rudder);
+
+    if (elevator == nullptr || aileron == nullptr || rudder == nullptr) {
```

Comment:
Magic! Basically rudder only planes still have ailerons :) (You don't do anything with the channel, but RC1 is still normally aileron.